### PR TITLE
Added theming

### PR DESCRIPTION
**Pull request by kristofdegrave**
This is my proposal for working with a theme

```
{
    theme: "bootstrap3",
    themes: {
       bootstrap3: {
             menu: '<ul class="typeahead dropdown-menu" role="listbox"></ul>',
             item: '<li><a class="dropdown-item" href="#" role="option"></a></li>',
             itemContentSelector: "a",
             headerHtml: '<li class="dropdown-header"></li>',
             headerDivider: '<li class="divider" role="separator"></li>'
       },     
       bootstrap4: {
             menu: "<div class=\"typeahead dropdown-menu\" role=\"listbox\"></div>",
             item: "<button class=\"dropdown-item\" role=\"option\"></button>",
             itemContentSelector: ".dropdown-item",
             headerHtml: "<h6 class=\"dropdown-header\"></h6>",
             headerDivider: "<div class=\"dropdown-divider\"></div>"
       } 
   }
}
```

so the default theme should be bootstrap3, you can choose bootstrap 4 if you want.

When setting the templates I would suggest the following logic:
1. check if 1 of the 5 properties (menu, item, ...) is provided use that.
2. in case that the above is undefined, use the provided theme
3. if the above should be undefined (= unvalid theme) fallback on bootstrap3

### CODE REVIEW

File: bootstrap3-typeahead.js [hunk 1]
```diff
@@ -61,7 +61,8 @@
     this.followLinkOnSelect = this.options.followLinkOnSelect || this.followLinkOnSelect;
     this.source = this.options.source;
     this.delay = this.options.delay;
-    this.$menu = $(this.options.menu);
+    this.theme = this.option.theme && this.options.themes && this.options.themes[this.options.theme] || Typeahead.defaults.themes[Typeahead.defaults.theme];
```
Comment (Chetane) [inline bootstrap3-typeahead.js]: @bassjobsen @kristofdegrave this seems to be a breaking change: `this.option` is undefined (missing letter `s`)

File: bootstrap3-typeahead.js [hunk 2]
```diff
@@ -61,7 +61,8 @@
...
+    this.$menu = $(this.options.menu || theme.menu);
```
Comment (Chetane) [inline bootstrap3-typeahead.js]: `theme` is also undefined, maybe `this.theme.menu`?

Comment (kristofdegrave) [reply bootstrap3-typeahead.js]: You are correct, I'll look into this asap an provide a new PR

File: bootstrap3-typeahead.js (hunk 1 above)
Comment (kristofdegrave) [reply bootstrap3-typeahead.js]: It should be `this.options` type (sorry :) )

Comment (kristofdegrave) [reply bootstrap3-typeahead.js]: Added PR #336

**Pull request merged**